### PR TITLE
Integrated the Grafana Dashboard into the operator chart

### PR DESCRIPTION
**Pull request by itay-grudev**
This adds the CloudNativePG Grafana Dashboard to the operator chart.

There are two flags:
* `enabled` (defaults to `false`) - whether to deploy the dashboard configmap
* `namespace` - allows overriding the namespace where the chart is located.

Example:
```console
helm install cnpg ./charts/cloudnative-pg \
  --namespace cnpg-system \
  --create-namespace \
  --set monitoring.grafanaDashboard.enabled=true \
  --set monitoring.grafanaDashboard.namespace=monitoring
```

### CODE REVIEW

File: charts/cloudnative-pg/templates/grafana-dashboard.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{{- if .Values.monitoring.grafanaDashboard.enabled -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cnpg-dashboard
+  namespace: {{ default .Release.Namespace .Values.monitoring.grafanaDashboard.namespace }}
+  labels:
+    grafana_dashboard: "1"
```
Comment (phisco) [inline charts/cloudnative-pg/templates/grafana-dashboard.yaml:8]: Label selectors can be [configured](https://github.com/grafana/helm-charts/blob/674b94682f1a2ebf8a72ca9a698f181b52c05592/charts/grafana/README.md?plain=1#L191), so please add a field to specify them defaulting to `grafana_dashboard`.

File: charts/cloudnative-pg/templates/grafana-dashboard.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{{- if .Values.monitoring.grafanaDashboard.enabled -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cnpg-dashboard
```
Comment (phisco) [inline charts/cloudnative-pg/templates/grafana-dashboard.yaml:5]: Can you make the name configurable through the values? maybe `cnpg-grafana-dashboard` as default just to be clearer.

**Comment by phisco** [comment]
Thanks, @itay-grudev! Can you take care of updating all the other references to point to this file instead of https://github.com/cloudnative-pg/cloudnative-pg/blob/main/docs/src/samples/monitoring/grafana-configmap.yaml? IIRC the original reason we removed it from here was because we had some issues keeping the two in sync.

File: charts/cloudnative-pg/templates/grafana-dashboard.yaml (hunk 1 above)
Comment (itay-grudev) [reply charts/cloudnative-pg/templates/grafana-dashboard.yaml:8]: I did not know that. Thanks a lot. I'll make the change.

**Comment by itay-grudev** [comment]
@phisco I'll make sure to submit the changes to `cloudnative-pg/cloudnative-pg` the moment this PR gets merge.

File: charts/cloudnative-pg/templates/grafana-dashboard.yaml (hunk 1 above)
Comment (itay-grudev) [reply charts/cloudnative-pg/templates/grafana-dashboard.yaml]: Changed to:

```yaml
  labels:
    {{ .Values.monitoring.grafanaDashboard.sidecarLabel }}: "1"
```

File: charts/cloudnative-pg/templates/grafana-dashboard.yaml (hunk 2 above)
Comment (itay-grudev) [reply charts/cloudnative-pg/templates/grafana-dashboard.yaml]: Changed to:

```yaml
name: {{ .Values.monitoring.grafanaDashboard.configMapName }}
```

with the new default value:

```yaml
configMapName: "cnpg-grafana-dashboard"
```

File: charts/cloudnative-pg/templates/grafana-dashboard.yaml [hunk 3]
```diff
@@ -2,10 +2,10 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: cnpg-dashboard
+  name: {{ .Values.monitoring.grafanaDashboard.configMapName }}
   namespace: {{ default .Release.Namespace .Values.monitoring.grafanaDashboard.namespace }}
   labels:
-    grafana_dashboard: "1"
+    {{ .Values.monitoring.grafanaDashboard.sidecarLabel }}: "1"
```
[the remark this answers is not in the document]
Comment (itay-grudev) [reply charts/cloudnative-pg/templates/grafana-dashboard.yaml:8]: I was just checking if that's the case.

File: charts/cloudnative-pg/templates/grafana-dashboard.yaml (hunk 1 above)
Comment (itay-grudev) [reply charts/cloudnative-pg/templates/grafana-dashboard.yaml]: Done.

File: charts/cloudnative-pg/templates/grafana-dashboard.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (itay-grudev) [reply charts/cloudnative-pg/templates/grafana-dashboard.yaml]: Done.

File: charts/cloudnative-pg/values.yaml [hunk 4]
```diff
@@ -139,6 +139,16 @@ affinity: {}
 monitoring:
   # -- Specifies whether the monitoring should be enabled. Requires Prometheus Operator CRDs.
   podMonitorEnabled: false
+  grafanaDashboard:
+    enabled: false
```
Comment (phisco) [inline charts/cloudnative-pg/values.yaml:143]: In the rest of the values where we allow specifying whether we have to create some resource from the helm chart we use `create`, `enabled` is more for stuff the operator will then handle:
```suggestion
    create: false
```

File: charts/cloudnative-pg/templates/grafana-dashboard.yaml (hunk 1 above)
Comment (itay-grudev) [reply charts/cloudnative-pg/templates/grafana-dashboard.yaml]: Changed to:
```yaml
  labels:
    {{ .Values.monitoring.grafanaDashboard.sidecarLabel }}: {{ .Values.monitoring.grafanaDashboard.sidecarLabelValue | quote }}
```

File: charts/cloudnative-pg/values.yaml [hunk 5]
```diff
@@ -139,6 +139,16 @@ affinity: {}
...
+    # -- Allows overriding the namespace where the dashboard is installed.
```
Comment (phisco) [inline charts/cloudnative-pg/values.yaml:144]:
```suggestion
    # -- Allows overriding the namespace where the ConfigMap will be created, defaulting to the same one as the Release.
```

File: charts/cloudnative-pg/values.yaml [hunk 6]
```diff
@@ -139,6 +139,16 @@ affinity: {}
...
+    namespace: ""
+    # -- The name of the configmap containing the dashboard.
```
Comment (phisco) [inline charts/cloudnative-pg/values.yaml:146]:
```suggestion
    # -- The name of the ConfigMap containing the dashboard.
```

File: charts/cloudnative-pg/values.yaml [hunk 7]
```diff
@@ -139,6 +139,16 @@ affinity: {}
...
+    configMapName: "cnpg-grafana-dashboard"
+    # -- Label that config maps with dashboards should have to be added
```
Comment (phisco) [inline charts/cloudnative-pg/values.yaml:148]:
```suggestion
    # -- Label that ConfigMaps should have to be loaded as dashboards.
```

File: charts/cloudnative-pg/values.yaml [hunk 8]
```diff
@@ -139,6 +139,16 @@ affinity: {}
...
+    sidecarLabel: "grafana_dashboard"
+    # -- Label value that config maps with dashboards should have to be added
```
Comment (phisco) [inline charts/cloudnative-pg/values.yaml:150]: I get you were only copying the upstream ones, but I prefer to be consistent with casing on resources names and it didn't sound right the upstream version.
```suggestion
    # -- Label value that ConfigMaps should have to be loaded as dashboards.
```

File: charts/cloudnative-pg/templates/grafana-dashboard.yaml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+{{- if .Values.monitoring.grafanaDashboard.enabled -}}
```
Comment (phisco) [inline charts/cloudnative-pg/templates/grafana-dashboard.yaml:1]: Here it should be updated to create too

**Review by phisco — APPROVED** [review]
Thanks @itay-grudev!

**Review by sxd — APPROVED** [review]

**Review by gbartolini — APPROVED** [review]

**Pull request merged**